contrib/dracut/90zfs/zfs-generator.sh.in

```diff
@@ -23,12 +23,11 @@ type getarg >/dev/null 2>&1 || {
 # If root is not ZFS= or zfs: or rootfstype is not zfs
 # then we are not supposed to handle it.
 [ "${root##zfs:}" = "${root}" -a "${root##ZFS=}" = "${root}" -a "$rootfstype" != "zfs" ] && exit 0
-# If root is set to zfs:AUTO, then we are also not
-# supposed to handle it, and it should be handled
-# by the traditional Dracut mount hook.
-# See https://github.com/zfsonlinux/zfs/pull/4558#discussion_r61118952
 if [ "${root}" = "zfs:AUTO" ] ; then
-  exit 0
+  root=$(zpool list -H -o bootfs)
```

Comment:
This STILL doesn't work, but at least it's closer, for some reason sysroot.mount isn't being run.  I can manually start it just fine though.  I'd say that this is better than the previous situation where sysroot.mount had `mount zfs:AUTO -o ro` though.